source/2_platform/rex_opengl/include/rex_opengl/resources/shader_program.h

```diff
@@ -0,0 +1,89 @@
+#pragma once
+
+#include "rex_opengl/resources/resource.h"
+
+#include "rex_engine/types.h"
+
+#include "rex_std/unordered_map.h"
+#include "rex_std/bonus/string/stack_string.h"
+
+#include <glm/glm.hpp>
+
+namespace rex
+{
+    class ShaderUniformLocation
+    {
+    public:
+        static ShaderUniformLocation s_empty;
+
+        //-------------------------------------------------------------------------
+        ShaderUniformLocation()
+            :m_location(-1)
+        {}
+        //-------------------------------------------------------------------------
+        ShaderUniformLocation(s32 loc)
+            :m_location(loc)
+        {}
+
+        //-------------------------------------------------------------------------
+        operator bool() const
+        {
+            return m_location != -1;
+        }
+        //-------------------------------------------------------------------------
+        operator s32() const
```

Comment:
Please verify that if you'd use the code as the following if (uniform_location) which conversion it'd use, the bool one or the s32 one